### ISSUE TITLE
[Dask-on-Ray][Bug] Ray remote resources having unexpected {'CPU': 1.0}

### ISSUE THREAD

**Issue by mlguruz** [ray-project/ray#23467]
### Search before asking

- [X] I searched the [issues](https://github.com/ray-project/ray/issues) and found no similar issues.

### Ray Component

Ray Core, Others

### Issue Severity

High: It blocks me to complete my task.

### What happened + What you expected to happen
Related to https://github.com/ray-project/ray/issues/21536

Ray remote resources having extra info after passing task-wise resource requirements thru `dask.annotate` (see below detailed example)

### Versions / Dependencies

Python = 3.9.12
Dask = 2022.03.0
Ray = 2.0.0.dev0

### Reproduction script

Run below:
```
import ray
ray.init()

import dask
from dask import delayed 
from ray.util.dask import ray_dask_get

def f1():
    return 1

def f2():
    return 2

@delayed 
def f3(X, y):
    return X + y

def run(x):
    X, y = f1(), f2()
    with dask.annotate(ray_remote_args=dict(resources={"num_cpus": 2})):
        m1 = f3(X, y)
    return m1

run(2).compute(scheduler=ray_dask_get)
```

, and you'll see:

```
(scheduler +6s) Tip: use `ray status` to view detailed cluster status. To disable these messages, set RAY_SCHEDULER_EVENTS=0.
(scheduler +6s) Error: No available node types can fulfill resource request {'num_cpus': 2.0, 'CPU': 1.0}. Add suitable node types to this cluster to resolve this issue.
(scheduler +41s) Error: No available node types can fulfill resource request {'CPU': 1.0, 'num_cpus': 2.0}. Add suitable node types to this cluster to resolve this issue.
(scheduler +1m16s) Error: No available node types can fulfill resource request {'num_cpus': 2.0, 'CPU': 1.0}. Add suitable node types to this cluster to resolve this issue.
(scheduler +1m51s) Error: No available node types can fulfill resource request {'num_cpus': 2.0, 'CPU': 1.0}. Add suitable node types to this cluster to resolve this issue.
(scheduler +2m26s) Error: No available node types can fulfill resource request {'CPU': 1.0, 'num_cpus': 2.0}. Add suitable node types to this cluster to resolve this issue.
(scheduler +3m1s) Error: No available node types can fulfill resource request {'CPU': 1.0, 'num_cpus': 2.0}. Add suitable node types to this cluster to resolve this issue.
```

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

**Comment by scottsun94** [comment]
Is this fixed now? Have you tried ray 2.0? @mlguruz

**Comment by mlguruz** [comment]
> Is this fixed now? Have you tried ray 2.0? @mlguruz 

Yes it works - I'm on 3.0.0dev now actually :) 

Feel free to close it. Thanks
